share/blob_tx.go

```diff
@@ -0,0 +1,78 @@
+package share
+
+import (
+	"errors"
+
+	v1 "github.com/celestiaorg/go-square/proto/blob/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	// ProtoBlobTxTypeID is included in each encoded BlobTx to help prevent
+	// decoding binaries that are not actually BlobTxs.
+	ProtoBlobTxTypeID = "BLOB"
+)
+
+type BlobTx struct {
+	Tx    []byte
+	Blobs []*Blob
+}
+
+// UnmarshalBlobTx attempts to unmarshal a transaction into blob transaction. If an
+// error is thrown, false is returned.
```

Comment:
IMO the second sentence is a bit misleading b/c this function can return an error and isBlob=false

```suggestion
// UnmarshalBlobTx attempts to unmarshal a transaction into blob transaction.
```